packages/vite-plugin-cloudflare/src/asset-config.ts

```diff
@@ -0,0 +1,128 @@
+import * as path from "node:path";
+import {
+	constructHeaders,
+	constructRedirects,
+} from "@cloudflare/workers-shared/utils/configuration/constructConfiguration";
+import { parseHeaders } from "@cloudflare/workers-shared/utils/configuration/parseHeaders";
+import { parseRedirects } from "@cloudflare/workers-shared/utils/configuration/parseRedirects";
+import {
+	HEADERS_FILENAME,
+	REDIRECTS_FILENAME,
+} from "@cloudflare/workers-shared/utils/constants";
+import { maybeGetFile } from "@cloudflare/workers-shared/utils/helpers";
+import {
+	HeadersSchema,
+	RedirectsSchema,
+} from "@cloudflare/workers-shared/utils/types";
+import { Log } from "miniflare";
+import type { ResolvedPluginConfig } from "./plugin-config";
+import type { ResolvedConfig } from "vite";
+import type { Unstable_Config } from "wrangler";
+
+/**
+ * Returns true if the `changedFile` matches one of the _headers or _redirects files,
+ * and the experimental support for these files is turned on.
+ */
+export function hasAssetsConfigChanged(
+	resolvedPluginConfig: ResolvedPluginConfig,
+	resolvedViteConfig: ResolvedConfig,
+	changedFile: string
+) {
+	if (!resolvedPluginConfig.experimental?.headersAndRedirectsDevModeSupport) {
+		return false;
+	}
+	return [
+		getRedirectsConfigPath(resolvedViteConfig),
+		getHeadersConfigPath(resolvedViteConfig),
+	].includes(changedFile);
+}
+
+/**
+ * Computes the assets config that will be passed to Miniflare,
+ * taking into account whether experimental _headers and _redirects support is on.
+ */
+export function getAssetsConfig(
+	resolvedPluginConfig: ResolvedPluginConfig,
+	entryWorkerConfig: Unstable_Config | undefined,
+	viteLogger: Log,
```

Comment:
Why are we importing `Log` from Miniflare and then calling it `viteLogger`. Can't we use the Vite logger directly?

Reply:
So strictly its a `ViteMiniflareLogger`. So it is a bit of both 😄 . Can rename.

Reply:
Or remove! 5430df921ca18ed5497f4b19ec58307f08d89ace